### PR TITLE
docs: document onboarding ids rules

### LINKED ISSUE #4911: Apply naming pattern for onboarding ids
### Is your enhancement related to a problem? Please describe

The ids used during onboarding aren't especially obvious when you look at logs or telemetry data, e.g. picking on compose:

- checkComposeDownloaded - IMHO it checks for 'local install'
- startComposeDownload - doesn't actually download, it just tells the user it will
- composeDownloaded is the success equivalent of composeFailedDownload

### Describe the solution you'd like

We should define some simple patterns that make onboarding ids consistent and fairly obvious, and apply it to both podman and compose extensions. It should have:

- clear differentiation between (a pattern for) ids for content vs command
- similar pattern for success vs failure screens (e.g. '.success' and '.failure')
- pattern for welcome or summary content pages

Do we need the extension in the id? (e.g. do compose ids need to include 'compose' or is that always obvious in context and could be removed?)

It should be obvious from the ids what the content shows or command does.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

## Rules
- steps displaying information are suffixed with `View`
- steps executing a command are suffixed with `Command`
- steps with a state='failed' are suffixed with `Failure`
- steps with a state='completed' are suffixed with `Success`

## Acceptance criteria

- [x] modify Podman onboarding (#4935)
- [x] Modify Compose onboarding (#4948)
- [ ] Update documentation to describe the rules (blocked by PR 4837)
- [x] Raise warnings when extension is loaded if rules are not respected (#4951)

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What issues does this PR fix or reference?

Fixes #4911

### CODE REVIEW

File: website/docs/extensions/write/onboarding-workflow.md [hunk 1]
```diff
@@ -222,6 +222,16 @@ A step has only two mandatory fields - id and title. All other properties are op
 
 The **id** must be unique to identify a step, and it is never displayed directly to the user.
 
+To analyse more easily in telemetry the steps executed by users, the **id** values must respect some rules.
```
Comment (benoitf) [inline website/docs/extensions/write/onboarding-workflow.md]:
```suggestion
To analyze more easily in telemetry the steps executed by users, the **id** values must respect some rules.
```
(just b/c target is American english :-)

**Review by benoitf — APPROVED** [review]
I don't know which editor you're using but you could install vale plug-in so you could have vale reports directly inline

**Comment by feloy** [comment]
> I don't know which editor you're using but you could install vale plug-in so you could have vale reports directly inline

I'm trying to make it work in vscode, but I get this error:

```
E100 [loadStyles] Runtime error

style 'RedHat' does not exist on StylesPath
```

Looking at .vale.ini, I can see styles should be in .vale/styles, where I can see only PodmanDesktop style, no RedHat one

Should I install it locally?

**Comment by feloy** [comment]
> > I don't know which editor you're using but you could install vale plug-in so you could have vale reports directly inline
> 
> I'm trying to make it work in vscode, but I get this error:
> 
> ```
> E100 [loadStyles] Runtime error
> 
> style 'RedHat' does not exist on StylesPath
> ```
> 
> Looking at .vale.ini, I can see styles should be in .vale/styles, where I can see only PodmanDesktop style, no RedHat one
> 
> Should I install it locally?

Finally I found it here: https://github.com/redhat-documentation/vale-at-red-hat

**Comment by benoitf** [comment]
@feloy you need to run 'vale sync' or `yarn run vale` in `website` folder

**Pull request merged**